### PR TITLE
Logging behavior doc

### PR DESCRIPTION
**Pull request by maridematte**
Fixes https://github.com/dotnet/msbuild/issues/10676

### Context
There has some confusion about which logs display what type of message. This is to clarify what is to be expected from each logger and verbosity.

### CODE REVIEW

File: documentation/Logging-behaviour.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+If you are expecting to see a certain type of message (like test logs) but are unable to find it in one of our loggers, check if the verbosity is correct and if the message has the correct type and importance.
+
+### Message types
+There are various types of messages within MSBuild with different importances and purposes. .
+There are some message types that are built-in within the engine, `errors`, `warnings`, and MSBuild engine information. The custom messages, that can come either from the engine or other sources, are selected and displayed based on the `importance` of the message. There can be high, normal, and low importance messages being displayed.
+
+For more information on custom messages you can more ou Microsoft Learn page for the [MSBuild Message Task](https://learn.microsoft.com/visualstudio/msbuild/message-task)
```
Comment (JanKrivanek) [inline documentation/Logging-behaviour.md:10]: Words order got broken here :-)

File: documentation/Logging-behaviour.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+## Logging behaviour
+MSBuild has a few different built-in loggers, which have different behaviours depending on verbosity. For more information on loggers you can visit the [Microsoft Learn page](ttps://learn.microsoft.com/visualstudio/msbuild/obtaining-build-logs-with-msbuild), or take a look at the [high-level overview of MSBuild](https://github.com/dotnet/msbuild/blob/main/documentation/High-level-overview.md#diagnosability--loggers).
+
+If you are expecting to see a certain type of message (like test logs) but are unable to find it in one of our loggers, check if the verbosity is correct and if the message has the correct type and importance.
+
+### Message types
+There are various types of messages within MSBuild with different importances and purposes. .
```
Comment (JanKrivanek) [inline documentation/Logging-behaviour.md:7]: nit:

```suggestion
There are various types of messages within MSBuild with different importances and purposes.
```

File: documentation/Logging-behaviour.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+There are some message types that are built-in within the engine, `errors`, `warnings`, and MSBuild engine information. The custom messages, that can come either from the engine or other sources, are selected and displayed based on the `importance` of the message. There can be high, normal, and low importance messages being displayed.
```
Comment (JanKrivanek) [inline documentation/Logging-behaviour.md:8]: The displaying based on importance applies only for Console logger - so maybe that part of the information should be moved to that section?

File: documentation/Logging-behaviour.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+For more information on custom messages you can more ou Microsoft Learn page for the [MSBuild Message Task](https://learn.microsoft.com/visualstudio/msbuild/message-task)
+
+### Console logger
+Console logger refers to the logger that outputs to the consolse in VS or the terminal that is being used. It is not the default logger after the [.NET 9 update](https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/9.0/terminal-logger) but still can be accessed by opting out of the Terminal Logger feature.
```
Comment (JanKrivanek) [inline documentation/Logging-behaviour.md:13]:
```suggestion
Console logger refers to the logger that outputs to the console in VS or the terminal that is being used. It is not the default logger after the [.NET 9 update](https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/9.0/terminal-logger) but still can be accessed by opting out of the Terminal Logger feature.
```

File: documentation/Logging-behaviour.md (hunk 3 above)
Comment (JanKrivanek) [reply documentation/Logging-behaviour.md:8]: Oh I see you have nice tables for that below - so maybe just mention that here (something like 'detailed overview of what gets displayed is listed per individual loggers in matching sections')

File: documentation/Logging-behaviour.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+For other messages the display follows the table below:
+| Verbosity                  | Quiet | Minimal | Normal | Detailed | Diagnostic |
+| ---------                  | ----- | ------- | ------ | -------- | ---------- |
+| Errors                     |&check;|
+| Warnings                   |&check;|
+| High-importance messages   |
+| Normal-importance messages |
+| Low-importance messages    |
+| MSBuild engine information |
```
Comment (JanKrivanek) [inline documentation/Logging-behaviour.md:37]: This seem to miss cases
https://github.com/dotnet/msbuild/pull/9810#issuecomment-1976663723

**Review by donJoseLuis — APPROVED** [review]

File: documentation/Logging-behaviour.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+### Text logger
+The text logger saves all the build data to a file. It's verbosity is determined by passing the `verbosity` parameter to the command, and it follows the same message display rules as the console logger.
```
Comment (JanKrivanek) [inline documentation/Logging-behaviour.md:40]: I believe (but I'm very unsure), that the standalone verbosity parameter applies only to console logger. To file logger it needs to be specified specifically for that logger - via `/flp:v-...` - similarly logger specific verbosity can be specified for terminallogger (`/tlp`) and console logger (`/clp`) as well

File: documentation/Logging-behaviour.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+| Verbosity                  | Quiet | Minimal | Normal | Detailed | Diagnostic |
+| ---------                  | ----- | ------- | ------ | -------- | ---------- |
+| Errors                     |&check;|
+| Warnings                   |&check;|
+| High-importance messages   |
+| Normal-importance messages |
+| Low-importance messages    |
+| MSBuild engine information |
+
+### Text logger
```
Comment (JanKrivanek) [inline documentation/Logging-behaviour.md:39]: Should this be named File logger?

**Review by JanKrivanek — COMMENTED** [review]
This looks very helpful!!

File: documentation/Logging-behaviour.md (hunk 5 above)
Comment (maridematte) [reply documentation/Logging-behaviour.md:37]: Yeah, I was not sure what the cases were here so I left it open. I just tested it manually and filled the table now.

File: documentation/Logging-behaviour.md (hunk 6 above)
Comment (maridematte) [reply documentation/Logging-behaviour.md:40]: The standalone applies to console logger, and terminal logger, but not for the File logger. Just tested it all manually with a sample project.

File: documentation/Logging-behaviour.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+| High-importance messages   |       |         |        | &check;  |   &check;  |
+| Normal-importance messages |
+| Low-importance messages    |
+| MSBuild engine information |
+
+### File logger
+The File logger saves all the build data to a file. It's verbosity is determined by passing the `verbosity` parameter to the `flp` attribute, or the default is set to `diagnostic`, and it follows the same message display rules as the console logger.
+
+### Binary logger / build logger
+The binary logger is a bit different as it does not have a verbosity option. It includes all messages, projects and files from the build by default.
```
Comment (JanProvaznik) [inline documentation/Logging-behaviour.md:45]:
```suggestion
The binary logger does not have a verbosity option. It includes all messages, projects and files from the build by default.
```

File: documentation/Logging-behaviour.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+| Verbosity                  | Quiet | Minimal | Normal | Detailed | Diagnostic |
+| ---------                  | ----- | ------- | ------ | -------- | ---------- |
+| Errors                     |&check;| &check; | &check;| &check;  |   &check;  |
+| Warnings                   |&check;| &check; | &check;| &check;  |   &check;  |
+| High-importance messages   |       |         |        | &check;  |   &check;  |
+| Normal-importance messages |
+| Low-importance messages    |
+| MSBuild engine information |
+
+### File logger
```
Comment (JanProvaznik) [inline documentation/Logging-behaviour.md:41]: I'd reorder to make it correspond with how often we'd expect people use the different loggers: 
1. TL
2. BL
3. CL
4. FL (it references CL so they should be together imo)

File: documentation/Logging-behaviour.md [hunk 10]
```diff
@@ -9,20 +9,6 @@ There are some message types that are built-in within the engine, `errors`, `war
 
 For more information on custom messages you can more ou Microsoft Learn page for the [MSBuild Message Task](https://learn.microsoft.com/visualstudio/msbuild/message-task)
```
Comment (baronfel) [inline documentation/Logging-behaviour.md:10]:
```suggestion
For more information on custom messages please reference the Microsoft Learn page for the MSBuild [Message](https://learn.microsoft.com/visualstudio/msbuild/message-task) Task.
```

File: documentation/Logging-behaviour.md [hunk 11]
```diff
@@ -38,9 +24,24 @@ To specify verbosity the `-verbosity` flag or `/tlp:verbosity={verbosity level}`
 | Low-importance messages    |
 | MSBuild engine information |
 
+### Binary logger / build logger
+The binary logger does not have a verbosity option. It includes all messages, projects and files from the build by default.
```
Comment (baronfel) [inline documentation/Logging-behaviour.md:28]:
```suggestion
The binary logger does not have a verbosity option. It includes all messages, projects and files from the build by default. It is intended to be a tooling-friendly way to get detailed information about what happened during a build, for analysis or automated processing.
```

File: documentation/Logging-behaviour.md [hunk 12]
```diff
@@ -9,20 +9,6 @@ There are some message types that are built-in within the engine, `errors`, `war
...
-| ---------                  | ----- | ------- | ------ | -------- | ---------- |
-| Errors                     |&check;| &check; | &check;| &check;  | &check;    |
-| Warnings                   |&check;| &check; | &check;| &check;  | &check;    |
-| High-importance messages   |       | &check; | &check;| &check;  | &check;    |
-| Normal-importance messages |       |         | &check;| &check;  | &check;    |
-| Low-importance messages    |       |         |        | &check;  | &check;    |
-| MSBuild engine information |       |         |        |          |            |
-
 ### Terminal logger
 Terminal logger is a new feature which improves the console experience.
```
Comment (baronfel) [inline documentation/Logging-behaviour.md:13]:
```suggestion
Terminal logger is a new feature which improves the console experience for end users by focusing the output on the diagnostics raised from a build for each project. It also allows users to see at-a-glance information about how the engine is building their projects at any time. It is opinionated and explicitly hides some build messages and output to deliver a more streamlined end-user experience. Users that need more detailed output should use the [console logger](#console-logger) or a [binary log](#binary-logger-build-logger) along with the [Structured Log Viewer](https://msbuildlog.com/) to drive their investigations.
```

**Review by baronfel — APPROVED** [review]
I left a bit of wording suggestions, but overall I'm very happy with having a centralized place to put logger comparisons!

**Comment by JanProvaznik** [comment]
nit: behaviour is standard British English, to be consistent we should use American (behavior)

**Pull request merged**